src/scripts/main.js

```diff
@@ -1,3 +1,285 @@
+    });
+  }
+});
+
+formFields.forEach((field) => {
+  const normalizeName = field.toLocaleLowerCase();
+  const label = document.createElement('label');
+
+  label.textContent = field + ': ';
+
+  if (field === 'Office') {
+    const select = document.createElement('select');
+
+    select.setAttribute('name', `${normalizeName}`);
+
+    offices.forEach((office) => {
+      const option = document.createElement('option');
+
+      option.textContent = office;
+      select.append(option);
+    });
+
+    select.setAttribute('data-qa', `${normalizeName}`);
+    select.setAttribute('required', true);
+    label.append(select);
+    fragment.append(label);
+  } else {
+    const input = document.createElement('input');
+
+    input.setAttribute('data-qa', `${normalizeName}`);
+    input.setAttribute('name', `${normalizeName}`);
+    input.setAttribute('required', true);
+
+    if (field === 'Age' || field === 'Salary') {
+      input.setAttribute('type', 'number');
+    } else {
+      input.setAttribute('type', 'text');
+    }
+
+    if (field === 'Age') {
+      input.setAttribute('min', 18);
+      input.setAttribute('max', 90);
+    }
+
+    label.append(input);
+    fragment.append(label);
+  }
+});
+
+const button = document.createElement('button');
+
+button.textContent = 'Save to table';
+
+fragment.append(button);
+form.append(fragment);
+document.body.firstElementChild.after(form);
+
+const formBtn = form.querySelector('button');
+
+function addSeparator(num) {
+  return num.toLocaleString('en-US');
+}
+
+function CustomNotification(title, description, type) {
+  this.title = title;
+  this.description = description;
+  this.type = type;
+}
+
+const pushNotification = ({ title, description, type }) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', `${type}`);
+  notification.style.top = 15 + 'px';
+  notification.style.right = 15 + 'px';
+
+  const fragmentNot = document.createDocumentFragment();
+  const titleNot = document.createElement('h2');
+
+  titleNot.classList.add('title');
+  titleNot.textContent = title;
+
+  const descriptionNot = document.createElement('p');
+
+  descriptionNot.textContent = description;
+
+  fragmentNot.append(titleNot);
+  fragmentNot.append(descriptionNot);
+  notification.append(fragmentNot);
+
+  document.body.append(notification);
+
+  setTimeout(() => {
+    notification.hidden = true;
+  }, 2000);
+};
+
+formBtn.addEventListener('click', (e) => {
+  e.preventDefault();
+
+  const formValues = {
+    employeeName: form.elements.name.value,
+    position: form.elements.position.value,
+    office: form.elements.office.value,
+    age: form.elements.age.value,
+    salary: form.elements.salary.value.trim(),
+  };
+
+  const formSalaryValue =
+    !formValues.salary || isNaN(formValues.salary)
+      ? ''
+      : '$' + addSeparator(+formValues.salary);
+
+  const fields = validateFormFields(formValues);
```

Comment:
The `validateFormFields` function does not validate the 'Office' field. Consider adding a validation check for the 'Office' field to ensure it is not empty or invalid.